File: Day27/minMeetingRooms.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var minMeetingRooms = function (intervals) {
    let start = intervals.slice().sort((a, b) => a[0] - b[0]);
    let end = intervals.slice().sort((a, b) => a[1] - b[1]);
    console.log(start);
    console.log(end)

    let room = 0;
    let j = 0;

    for (let i = 0; i < start.length; i++) {
        if (start[i][0] < end[j][1]) {
            room++
        } else {
            j++
        }
    }
    return room;
};

File: Day27/minMeetingRooms.cpp
class Solution
{
public:
    int minMeetingRooms(vector<vector<int>> &intervals)
    {

        vector<int> x(intervals.size()); //start time
        vector<int> y(intervals.size()); //end time

        for (auto i = 0; i < intervals.size(); i++)
        {
            x[i] = intervals[i][0];
            y[i] = intervals[i][1];
        }

        //sort
        sort(begin(x), end(x));
        sort(begin(y), end(y));

        int room = 0;
        int j = 0;

        for (int i = 0; i < intervals.size(); i++)
        {
            //start time is less than end time that means the metting is not over yet
            if (x[i] < y[j])
            {
                room++;
            }
            else
            {
                j++;
            }
        }

        return room;
    }
};